plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,257 @@
+end
+
+function AutoDim:getAutodimMenu()
+    return {
+        text = _("Automatic dimmer"),
+        checked_func = function() return self.autodim_starttime_m > 0 end,
+        sub_item_table = {
+            {
+                text_func = function()
+                    return self.autodim_starttime_m <= 0 and _("Idle time for dimmer") or
+                    T(_("Idle time for dimmer: %1 min"), self.autodim_starttime_m)
+                end,
+                checked_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local idle_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer idle time"),
+                        info_text = _("Start the dimmer after the designated period of inactivity."),
+                        value = self.autodim_starttime_m >=0 and self.autodim_starttime_m or 0.5,
+                        default_value = DEFAULT_AUTODIM_STARTTIME_M,
+                        value_min = 0.5,
+                        value_max = 60,
+                        value_step = 0.5,
+                        value_hold_step = 5,
+                        unit = _("min"),
+                        precision = "%0.1f",
+                        ok_always_enabled = true,
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_starttime_m = spin.value
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function()
+                            self.autodim_starttime_m = -1
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", -1)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(idle_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimmer duration: %1 s"), self.autodim_duration_s)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local dimmer_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer duration"),
+                        info_text = _("Enter the duration until reaching the final brightness."),
```

Comment:
Do we use such an imperative verb elsewhere ("Enter ...!") ?
Just "Delay to reach the final brightness" (a description, with an alternative wording, using different words, to make this possibly less ambiguous for translators and users) ?

Reply:
In the `DateTimeWidget` we used `Enter a time in ...`. 
I will change that later (here and in the DTW Pr).